sdk/core/azure-core/test/ut/CMakeLists.txt

```
@@ -88,6 +88,10 @@ add_executable (
   azure-core-global-context-test
   global_context.cpp
 )
+if (MSVC)
+  # Disable gtest warnings for msvc
+  target_compile_options(azure-core-global-context-test PUBLIC /wd26495 /wd26812 /wd6326 /wd28204 /wd28020 /wd6330 /wd4389)
```

What's causing these warnings?

---

Nevermind, saw some from the issue:
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=775326&view=logs&j=32afe1f3-cae1-5682-187c-9a515535777f&t=83f6091f-1e1e-5cf9-2a4c-1792f2cefa30&s=4ff3786a-ea67-5356-a1c6-e7ae13aeb7e1

```text
2021-03-10T04:18:56.0178729Z D:\a\_work\1\s\build\_deps\googletest-src\googletest\include\gtest/internal/gtest-internal.h(605,1): error C2220: the following warning is treated as an error [D:\a\_work\1\s\build\sdk\core\azure-core\test\ut\azure-core-global-context-test.vcxproj]
2021-03-10T04:18:56.0184272Z D:\a\_work\1\s\build\_deps\googletest-src\googletest\include\gtest\internal\gtest-internal.h(603): warning C6326: Potential comparison of a constant with another constant. [D:\a\_work\1\s\build\sdk\core\azure-core\test\ut\azure-core-global-context-test.vcxproj]
```

---

Do we need to skip all these warnings? Why not just 6326?
```suggestion
  target_compile_options(azure-core-global-context-test PUBLIC /wd6326)
```

---

Yes, since we are turning the compiler flags for warnings to be very strict, some of the google test comparisons becomes a warning, for instance, inferring types like `EXPECT_EQ(1, vector.size())`  .  Since vector returns a size_t and then we compared to a int, mscv generates a warning.
In the past I started to change all checks to
```cpp
site_t expected =1;
EXPECT_EQ(expected, vector.size());
```
But it makes things more complicated just for the tests, so I better just disable those common warnings from MSVC for test cases